docs/book/v2/quick-start.md

```diff
@@ -0,0 +1,149 @@
+# Quick Start
+
+## Mapping Routes to Middleware
+
+The first step is to map a route to PSR-7 middleware. This looks like any other
+[routing](https://docs.laminas.dev/laminas-mvc/routing/) configuration, with small changes: the `controller` key in the
+routing options has to be the `Laminas\Mvc\Middleware\PipeSpec` class literal, and you provide a `middleware` key.
+
+For example, to register an `AlbumListHandler` located in the `module/Application/Handler` directory to the routes of
+your `Application` module, add the following route to `module/Application/config/module.config.php`:
+
+```php
+use Application\Handler\AlbumListHandler;
+use Laminas\Mvc\Middleware\PipeSpec;
+use Laminas\Router\Http\Literal;
+
+return [
+    'router' => [
+        'routes' => [
+            'album-list' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/albums',
+                    'defaults' => [
+                        'controller' => PipeSpec::class,
+                        'middleware' => AlbumListHandler::class,
+                    ],
+                ],
+            ],
+        ],
+    ],
+];
+```
+
+Middleware may be provided as instance of a PSR-15 `Psr\Http\Server\MiddlewareInterface`
+or `Psr\Http\Server\RequestHandlerInterface`
+or as service name strings resolving to such instances.  
+You may also specify an instance of the `PipeSpec` class which accepts both middleware types above or their service name
+strings, or a `Closure` . These will then be piped into a `Laminas\Stratigility\MiddlewarePipe` instance in the order in
+which they are present in the `PipeSpec`. See [routing middleware](routing-middleware.md) for examples.
+
+> ### No Action Required
+>
+> Unlike action controllers, middleware typically is single purpose, and, as
+> such, does not require a default `action` parameter.
+
+## Middleware Services
+
+Middleware are pulled from the application service manager, unlike controllers in a normal
+[laminas-mvc](https://docs.laminas.dev/laminas-mvc/) dispatch cycle, which are pulled from a
+dedicated `ControllerManager`.
+
+Middleware retrieved **must** be a PSR-15 `MiddlewareInterface` or `RequestHandlerInterface` instance.
+Otherwise, `MiddlewareListener` will create an error response.
+
+## Writing Middleware
+
+The next step is writing actual middleware to dispatch. PSR-15 defines two different interfaces:
+
+### `MiddlewareInterface` vs. `RequestHandlerInterface`
+
+_Middleware_ is code sitting between a request and a response; it typically analyzes the request to aggregate incoming data, delegates it to another layer to process, and then creates and returns a response.
+
+A _RequestHandler_ is a class that receives a request and returns a response, without delegating to other layers of the application. This is generally the inner-most layer of your application.
```

Comment:
This is not entirely correct.

Request handlers are generally either the _inner-most_ **or** _outer-most_ layers of your application. When they are the outer-most layer, they **do** delegate to other layers of the application; as an example, in Mezzio, the application pipeline... is a request handler, and it dispatches the pipeline it composes in order to create a response.

Reply:
Hmm, to be honest, I borrowed that explanation from the [Mezzio docs](https://docs.mezzio.dev/mezzio/v3/getting-started/features/). Do you want me to change the wording? The "generally" in the explanation somehow suggest that there are also other ways. And as far as I know, `RequestHandler`s  cannot be used as outer-most layer in laminas-mvc.

Reply:
Yes, please change. If you have time, would you also please send a PR to the Mezzio docs to fix it there as well?

Technically, the outermost layer of the pipeline even in laminas-mvc-middleware is also a request handler, as it's a `MiddlewarePipe` instance... :smile:

Reply:
as discussed in [slack](https://laminas.slack.com/archives/CAP4J90LW/p1610557563136400), we leave that as-is.
Is there anything to be changed for the PR to be mergeable?